src/IECore/StreamIndexedIO.cpp

```diff
@@ -413,6 +413,11 @@ size_t compress(
 		return 0;
 	}
 
+	if ( compressionLevel == 0)
```

Comment:
Do the newly-added calls to `compress()` (when writing the index) cope with this early return? It looks like they unconditionally use `outputBuffer`. Is it possible to refactor to avoid this trap somehow?

Reply:
In 85010c9 removed the early out in `compress` and prevent the call to in the common data block case where our data isn't compressed. This allows for the possibility of a uncompressed index albeit wrapped in a blosc header and this uncommon case would have a redundant copy.